tools/icu/icutrim.py

```diff
@@ -285,14 +285,14 @@ def addTreeByType(tree, mytree):
         # read in the resource list for the tree
         treelistfile = os.path.join(options.tmpdir,"%s.lst" % tree)
         runcmd("iculslocs", "-i %s -N %s -T %s -l > %s" % (outfile, dataname, tree, treelistfile))
-        fi = open(treelistfile, 'rb')
-        treeitems = fi.readlines()
-        trees[tree]["locs"] = [treeitems[i].strip() for i in range(len(treeitems))]
-        fi.close()
-        if tree not in config.get("trees", {}):
-            print(" Warning: filter file %s does not mention trees.%s - will be kept as-is" % (options.filterfile, tree))
-        else:
-            queueForRemoval(tree)
+        with io.open(treelistfile, 'r', encoding='utf-8') as fi:
+            treeitems = fi.readlines()
+            trees[tree]["locs"] = [treeitems[i].strip() for i in range(len(treeitems))]
```

Comment:
```suggestion
            trees[tree]["locs"] = [line.strip() for line in treeitems]
```

Reply:
I agree with you the list comprehension can be simplified and do not follow the [PEP 0020](https://www.python.org/dev/peps/pep-0020/). 

Previously I have not try to enhance this code as is not part of the change. And to my point of view lot of think should be changed, such as: global variables, open file without context manager, no main function, executable code is some time  between two function (as l266-l271) and so on

I added your suggestion ;-)